varianteval/core/io/vcf.py

```diff
@@ -53,9 +61,136 @@ def vcf2callset(vcf_fname, filter_funcs=None):
+    context).
+    
+    :param is_first_pos: TRUE=first position in a VCF record; FALSE=last 
+    position;
+    :param sigma_multiple: if the record does not contain coordinates but just 
+    the STD, the procedure assumes an interval that extends this multiple of the
+    STD on both sides (2 or 3 captures most of a Gaussian);
+    :return: First element: quantity to be added to ``X`` to get the first 
+    position of the uncertainty interval (typically negative or zero). Second 
+    element: quantity to be added to ``X`` to get the last position of the 
+    uncertainty interval (typically positive or zero).
+    """
+    quantum: int
+    value: float	
+
+    if is_first_pos:
+        if VCF_CIPOS_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIPOS_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_START:
+            if string in record.info.keys():
+                value = float(record.info.get(string))
+                quantum = int(value*SIGMA_MULTIPLE)
+                return (-quantum, quantum)
+    else:
+        if VCF_CIEND_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIEND_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        if VCF_CILEN_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CILEN_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_END:
+            if string in record.info.keys():
+                value = float(record.info.get(string))
+                quantum = int(value*SIGMA_MULTIPLE)
+                return (-quantum, quantum)
+    return (0, 0)
+
+
+def get_confidence_interval_std(record: VariantRecord, is_first_pos: bool) -> float:
+    """
+    Returns the STD of the range of uncertainty around a position ``X`` 
+    (implicit from the context).
+    
+    :param is_first_pos: TRUE=first position in a VCF record; FALSE=last 
+    position.
+    """
+    if is_first_pos:
+        for string in VCF_STD_START:
+            if string in record.info.keys():
+                return float(record.info.get(string))
+    else:
+        for string in VCF_STD_END:
+            if string in record.info.keys():
+                return float(record.info.get(string))
+    return 0.0
+
+
+def ct2side(record: VariantRecord) -> tuple[int,int]:
+    """
+    Reads info about the CT field of a TRA record.
+    
+    :return: First element: left side. Second element: right side.
+    """
+    out: tuple[int,int]
+    
+    value = record.info.get(VCF_CT_STR)
+    if value == VCF_CT_325_STR:
+        out[0] = 1
+        out[1] = 0
+    elif value == VCF_CT_523_STR:
+        out[0] = 0
+        out[1] = 1
+    elif value == VCF_CT_525_STR:
+        out[0] = 0
+        out[1] = 0
+    elif value == VCF_CT_323_STR:
+        out[0] = 1
+        out[1] = 1
```

Comment:
is there an else?

Reply:
The else is not needed, but I forgot to initialize the tuple at the beginning. I improved the code by raising an exception instead.